src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -115,248 +119,73 @@ public void run()
                 }
             }
 
-
-            GetMethod httpget = null;
-            HeadMethod httphead = null;
+            HttpMethodBase httpMethod = null;
 
             try
             {
 
                 work = (WorkUnit) queue.take();
-                working = true;
+                working = true;	
                 url = work.getWork();
                 int code = 0;
 
-                String responce = "";
-                String rawResponce = "";
+                String response = "";
+                String rawResponse = "";
+                
+                httpMethod = getHttpMethod(work.getMethod(), url.toString());
 
                 //if the work is a head request
                 if(work.getMethod().equalsIgnoreCase("HEAD"))
                 {
-                    if(Config.debug)
-                    {
-                        System.out.println("DEBUG Worker[" + threadId + "]: HEAD " + url.toString());
-                    }
-
-                    httphead = new HeadMethod(url.toString());
-
-                    //set the custom HTTP headers
-                    Vector HTTPheaders = manager.getHTTPHeaders();
-                    for(int a = 0; a < HTTPheaders.size(); a++)
-                    {
-                        HTTPHeader httpHeader = (HTTPHeader) HTTPheaders.elementAt(a);
-                        /*
-                         * Host header has to be set in a different way!
-                         */
-                        if(httpHeader.getHeader().startsWith("Host"))
-                        {
-                            httphead.getParams().setVirtualHost(httpHeader.getValue());
-                        }
-                        else
-                        {
-                            httphead.setRequestHeader(httpHeader.getHeader(), httpHeader.getValue());
-                        }
-
-                    }
-                    httphead.setFollowRedirects(Config.followRedirects);
-
-                    /*
-                     * this code is used to limit the number of request/sec
-                     */
-                    if(manager.isLimitRequests())
-                    {
-                        while(manager.getTotalDone() / ((System.currentTimeMillis() - manager.getTimestarted()) / 1000.0) > manager.getLimitRequestsTo())
-                        {
-                            Thread.sleep(100);
-                        }
-                    }
-                    /*
-                     * Send the head request
-                     */
-                    code = httpclient.executeMethod(httphead);
+                    code = makeRequest(httpMethod);
                     if(Config.debug)
```

Comment:
The ```if``` block could/should be moved to ```makeRequest(HttpMethodBase)```, like it was the other debug statement, makes more sense to have both information "logged" (also some callers are not even "logging" which makes the previous "log" futile since we wouldn't know the outcome of the request).